modules/jax/src/main/ts/ephox/jax/core/HttpJwt.ts

```diff
@@ -10,24 +10,25 @@ const headers = (headersInput: HttpRequest<ResponseBodyDataTypes>['headers'], to
   return headersInput ? { ...headersInput, ...authHeader } : authHeader;
 };
 
-type RunMethod<T> = (token: JwtToken) => FutureResult<T, HttpError>;
+type RunMethod<T, U extends keyof ResponseTypeMap> = (token: JwtToken) => FutureResult<T, HttpError<U>>;
 
-const requestFreshToken = (tokenFactory: JwtTokenFactory): FutureResult<JwtToken, HttpError> => tokenFactory(true);
-const requestCachedToken = (tokenFactory: JwtTokenFactory): FutureResult<JwtToken, HttpError> => tokenFactory(false);
+const requestFreshToken = <T extends keyof ResponseTypeMap>(tokenFactory: JwtTokenFactory): FutureResult<JwtToken, HttpError<T>> => tokenFactory(true);
+const requestCachedToken = <T extends keyof ResponseTypeMap>(tokenFactory: JwtTokenFactory): FutureResult<JwtToken, HttpError<T>> => tokenFactory(false);
 
-const tryAgain = <T>(tokenFactory: JwtTokenFactory, runMethod: RunMethod<T>) => (error: HttpError) => error.status === HttpErrorCode.Unauthorized ? requestFreshToken(tokenFactory).bindFuture(runMethod) : FutureResult.error(error);
+const tryAgain = <T, U extends keyof ResponseTypeMap>(tokenFactory: JwtTokenFactory, runMethod: RunMethod<T, U>) =>
+  (error: HttpError<U>) => error.status === HttpErrorCode.Unauthorized ? requestFreshToken(tokenFactory).bindFuture(runMethod) : FutureResult.error(error);
 
-const runWithToken = <T extends keyof ResponseTypeMap>(runMethod: RunMethod<ResponseTypeMap[T]>, tokenFactory: JwtTokenFactory): FutureResult<T, HttpError> =>
+const runWithToken = <T extends keyof ResponseTypeMap>(runMethod: RunMethod<ResponseTypeMap[T], T>, tokenFactory: JwtTokenFactory): FutureResult<T, HttpError<any>> =>
```

Comment:
Haven't figured out how to get rid of `any` :/

Reply:
Just dumping this here from DM's, but I believe this should fix the `any`:
```
const tryAgain = <T, U extends keyof ResponseTypeMap>(tokenFactory: JwtTokenFactory, runMethod: RunMethod<T, U>) =>
  (error: HttpError<U>) => error.status === HttpErrorCode.Unauthorized ? requestFreshToken<U>(tokenFactory).bindFuture(runMethod) : FutureResult.error(error);

const runWithToken = <T extends keyof ResponseTypeMap>(runMethod: RunMethod<ResponseTypeMap[T], T>, tokenFactory: JwtTokenFactory): FutureResult<T, HttpError<T>> =>
  requestCachedToken<T>(tokenFactory)
    .bindFuture<ResponseTypeMap[T]>(
    (token) => runMethod(token).bind((result) => result.fold(tryAgain<T, T>(tokenFactory, runMethod), FutureResult.pure))
  );
```

Reply:
I've `Ctrl+c` `Ctrl+v` ed thanks.